tests/test_pytest.py

```diff
@@ -71,3 +71,23 @@ def test_z(): pass
     all_tests = [x.item.name for x in result.calls if x._name == 'pytest_runtest_call']
 
     assert set(group_1 + group_2) == set(all_tests)
+
+
+def test_random_group_runs_in_original_order(testdir):
+    """When running tests with a random seed, check test order is unchanged"""
+    testdir.makepyfile("""
+        def test_b(): pass
+        def test_c(): pass
+        def test_d(): pass
+        def test_e(): pass
+        def test_f(): pass
+        def test_g(): pass
+        def test_h(): pass
+        def test_i(): pass
+    """)
+
+    result = testdir.inline_run('--test-group-count', '2',
+                                '--test-group', '1',
+                                '--test-group-random-seed', '5')
+    group_1 = [x.item.name for x in result.calls if x._name == 'pytest_runtest_call']
+    assert group_1 == sorted(group_1)
```

Comment:
suggest reversing the test definitions above so that you can check it's sorted as-defined rather than alphabetical?

```
testdir.makepyfile("""
        def test_i(): pass
        def test_h(): pass
        ...
        def test_b(): pass
    """)
...
assert group_1 = sorted(group_1, reverse=True)
```